js/modules/k6/experimental/streams/errors.go

```diff
@@ -0,0 +1,64 @@
+package streams
+
+import "github.com/dop251/goja"
+
+func newError(k errorKind, message string) *streamError {
+	return &streamError{
+		Name:    k.String(),
+		Message: message,
+		kind:    k,
+	}
+}
+
+//go:generate enumer -type=errorKind -output errors_gen.go
+type errorKind uint8
+
+const (
+	// TypeError is thrown when an argument is not of an expected type
+	TypeError errorKind = iota + 1
+
+	// RangeError is thrown when an argument is not within the expected range
+	RangeError
+
+	// RuntimeError is thrown when an error occurs that was caused by the JS runtime
+	// and is not likely caused by the user, but rather the implementation.
+	RuntimeError
+
+	// AssertionError is thrown when an assertion fails
+	AssertionError
+)
+
+type streamError struct {
+	// Name contains the name of the error
+	Name string `json:"name"`
+
+	// Message contains the error message
+	Message string `json:"message"`
```

Comment:
That's a really good idea! I may give it a try, I fully agree that, despite not blocking (most likely), it will probably make our lives easier in the long term

Reply:
It has been trickier than I initially thought, because sometimes we need to return these errors as Go `error`s, or keep them stored in-memory and `return`/`throw` them later. So, we need to either convert every signature into `any` instead of `error`, which would be messy, or un/wrap these errors, which is... meh, but feasible.

That said, I feel like like I'm having low inspiration these days, so if you have any other better approach, suggestions will be more than welcome.

Reply:
I concur with the points brought up by @joanlopez, refactoring the error mechanism here would be quite herculean. I would also much rather do that as a separate task, and even use this as an opportunity to return more standard/conventional JS errors to the runtime :)

I'll open an issue 👍🏻